percona/controller/pgcluster/controller.go

```diff
@@ -175,7 +195,15 @@ func (r *PGClusterReconciler) Reconcile(ctx context.Context, request reconcile.R
 		},
 	}
 
+	watcherCtx, watcherCancel := context.WithCancel(ctx)
```

Comment:
I'm not sure about this context, I worry about a possible memory leak. Maybe we should use a channel to stop watchers? wdyt @inelpandzic ?

Reply:
This is the way to do it. Just make sure, if necessary, to check `watcherCtx.Done()` to stop some action. Down the line you pass it to `client.List`, that will probable check this, but in case you want to stop the action even before that, check the `Done` channel.

Also consider should we have `defer watcherCancel()` in case `cr.DeletionTimestamp != nil` is not met, probably not from what I see.

Reply:
no we shouldn't defer it because then it'll cancel the context at the end of the Reconcile function.

the thing that worries me is that we create a new context in every reconcile without canceling it